src/App.jsx

```diff
@@ -6,25 +6,72 @@ import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
 
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  
+  const fetchData = async () => {
+  const options = {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+    },
+  };
+
+  const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+
+  try {
+    const response = await fetch(url, options);
+
+    if (!response.ok) {
+      const message = `Error has ocurred:
+                             ${response.status}`;
+      throw new Error(message);
+    }
+
+    const data = await response.json();
+    
+    const todos = todosFromAPI.records.map((todo) => {
+      const newTodo =  {
+          id: todo.id,
+          title: todo.fields.title
+        }
+
+        return newTodo
+
+    });
+
+    setTodoList(todos);
+    setIsLoading(false);
+  } catch (error) {
+    console.error(error.message);
+    setIsLoading(false);
+  }
+};
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  )
 
+useEffect(() => {
+const getAsyncTodoList = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    resolve({
+      data: {
+        todoList: []  
+      }
+    });
+  }, 2000); 
+});
+
+getAsyncTodoList
+.then(result=>{
+  setTodoList(result.data.todoList)
+  setIsLoading(false)
+});
+}, []);
```

Comment:
For this lesson, you would need to get rid of most of this and call the `fetchData` function inside of the useEffect.

Reply:
Finish up with PR 7 and then come back to this one. Feel free to reach out to me if you run into any issues